reactor-core/src/main/java/reactor/core/publisher/OptimizableOperator.java

```diff
@@ -35,7 +35,13 @@
 
 	/**
 	 * Allow delegation of the subscription by returning a {@link CoreSubscriber}, or force
-	 * subscription encapsulation by returning null. This can be used in conjunction with {@link #nextOptimizableSource()}
+	 * subscription encapsulation by returning null.
+	 * <br/>
+	 * Caller MUST catch exceptions and report them to the most relevant {@link org.reactivestreams.Subscriber},
+	 * generally using {@link Operators#reportThrowInSubscribe(CoreSubscriber, Throwable)},
```

Comment:
Looking at the current code base, some implementors of this API actually do something like
```java
Operators.error(actual, Operators.onOperatorError(e, actual.currentContext()));
```

Is that OK? (If not, to be dealt with in another PR, but wanted to raise the point)

Reply:
@ericbottard could you please point me at any place where `Operators.error(actual, Operators.onOperatorError` in `subscribe` is used instead of `reportThrowInSubscribe`?

Reply:
From "Show call hierarchy" in IJ:
```
MonoIgnorePublisher
FluxReplay
MonoDelayUntil
MonoFromPublisher
```

Reply:
That's because `source` is `Publisher`, and `subscribeOrReturn` is defined next to `subscribe()` + final and will never throw

Reply:
yeah, I saw that hence the `generally`. it is ok for now but would make sense in aligning in a separate issue I think.

Reply:
There is nothing to align, we intentionally did not change these 4 :)

Reply:
ok then 👍